src/Test/QuickCheck/Classes/Generic.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+#if MIN_VERSION_base(4,12,0)
+{-# LANGUAGE QuantifiedConstraints #-}
+#endif
+{-# OPTIONS_GHC -Wall #-}
+
+module Test.QuickCheck.Classes.Generic
+  (
+#if MIN_VERSION_base(4,6,0)
+    genericLaws
+  , genericLawsU
+#endif
+#if MIN_VERSION_base(4,12,0) 
+--  , generic1Laws
+#endif
+  ) where
+
+#if MIN_VERSION_base(4,6,0)
+import Control.Applicative
+import Data.Semigroup
+import Data.Monoid
+import GHC.Generics
+#if MIN_VERSION_base(4,9,0)
+import Data.Functor.Classes
+#endif
+import Data.Proxy (Proxy(Proxy))
+import Test.QuickCheck
+import Test.QuickCheck.Property (Property)
+
+import Test.QuickCheck.Classes.Common (Laws(..), Apply(..))
+
+-- | Tests the following properties:
+--
+-- [/Conjunction Idempotence/]
+--   @n .&. n ≡ n@
+-- [/Disjunction Idempotence/]
+--   @n .|. n ≡ n@
+-- [/Double Complement/]
+--   @complement (complement n) ≡ n@
+--
+-- /Note:/ This property test is only available when
+-- using @base-4.6@ or newer.
+genericLaws :: (Generic a, Eq a, Arbitrary a, Show a, Show (Rep a x), Arbitrary (Rep a x), Eq (Rep a x)) => Proxy a -> Proxy x -> Laws
+genericLaws pa px = Laws "Generic"
+  [ ("From-To idempotence", fromToIdempotence pa px)
+  , ("To-From idempotence", toFromIdempotence pa)
+  ]
+
+genericLawsU :: (Generic a, Eq a, Arbitrary a, Show a, Show (Rep a ()), Arbitrary (Rep a ()), Eq (Rep a ())) => Proxy a -> Laws
```

Comment:
I'm not sure I understand why both `genericLaws` and `genericLawsU` exist. Shouldn't the former be sufficient?

Reply:
It is true that the former is sufficient, but passing an addition `Proxy` argument seems out of line with the rest of the API, so I provide this. If you think this is misguided, please explain why and I may remove it.

Reply:
I thought on the off-chance that somehow `x` mattered to someone then it might make sense to allow this, but you're right, `from` and `to` don't care about `x`, so it might as well just be `()` always.